### PR TITLE
Add/analysis overview query

### PR DESCRIPTION
**Pull request by barshathakuri**
Addresses
- https://github.com/orgs/the-deep/projects/2/views/5?pane=issue&itemId=58082282
- https://github.com/the-deep/client/issues/2898
- https://github.com/the-deep/client/issues/2914
- https://github.com/the-deep/client/issues/2916

Depends on: https://github.com/the-deep/server/pull/1516

## Changes

- Fetch analysis overview client
- Fetch analysis summer client
- Fetch analysis detail client
- Fetch pillar summary

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: app/views/AnalysisDashboard/Analysis/AnalysisCloneModal/index.tsx [hunk 1]
```diff
@@ -85,11 +84,11 @@ function AnalysisCloneModal(props: Props) {
         setFieldValue,
         validate,
         setError,
-        setValue,
     } = useForm(schema, defaultFormValue);
 
     const error = getErrorObject(riskyError);
 
+    /*
```
Comment (AdityaKhatri) [inline app/views/AnalysisDashboard/Analysis/AnalysisCloneModal/index.tsx]: yo block nai hatauda huncha

File: app/views/AnalysisDashboard/Analysis/AnalysisCloneModal/index.tsx [hunk 2]
```diff
@@ -19,10 +19,9 @@ import {
     createSubmitHandler,
 } from '@togglecorp/toggle-form';
 
-import { useRequest, useLazyRequest } from '#base/utils/restRequest';
+import { useLazyRequest } from '#base/utils/restRequest';
```
Comment (AdityaKhatri) [inline app/views/AnalysisDashboard/Analysis/AnalysisCloneModal/index.tsx:22]: TODO chodi rakha na, remove garna cha bhanera tha huncha

File: app/views/AnalysisDashboard/Analysis/AnalysisPillar/index.tsx [hunk 3]
```diff
@@ -74,13 +71,11 @@ function AnalysisPillar(props: Props) {
         statusLabel = _ts('analysis', 'noAnalysisTagLabel');
     }
 
-    const disabled = pendingPillarDelete;
-
     const onDeleteConfirmClick = useCallback(() => {
-        onDelete(pillarId);
+        onDelete(Number(pillarId));
```
Comment (AdityaKhatri) [inline app/views/AnalysisDashboard/Analysis/AnalysisPillar/index.tsx]: Leave a TODO here as well

File: app/views/AnalysisDashboard/Analysis/AnalysisPillar/index.tsx [hunk 4]
```diff
@@ -74,13 +71,11 @@ function AnalysisPillar(props: Props) {
         statusLabel = _ts('analysis', 'noAnalysisTagLabel');
     }
 
-    const disabled = pendingPillarDelete;
```
Comment (AdityaKhatri) [inline app/views/AnalysisDashboard/Analysis/AnalysisPillar/index.tsx:77]: We need this, don't we?

File: app/views/AnalysisDashboard/Analysis/PillarList/index.tsx [hunk 5]
```diff
@@ -1,119 +1,67 @@
...
-            onDelete: triggerPillarDelete,
-            statements: data.analyticalStatements,
+            onDelete: onAnalysisPillarDelete,
+            statements: data.statements,
             pillarId: data.id,
-            analyzedEntries: data.analyzedEntries,
+            analyzedEntries: data.analyzedEntriesCount,
             projectId: activeProject,
             title: data.title,
-            pendingPillarDelete: pendingPillarDelete && data.id === deletePillarId,
```
Comment (AdityaKhatri) [inline app/views/AnalysisDashboard/Analysis/PillarList/index.tsx:107]: Are we moving this inside?

File: app/views/AnalysisDashboard/Analysis/index.tsx [hunk 6]
```diff
@@ -429,7 +427,7 @@ function Analysis(props: ComponentProps) {
             <ExpandableContainer
                 className={styles.pillarAnalyses}
                 headerClassName={styles.pillarAnalysesHeader}
-                heading={_ts('analysis', 'pillarAnalysisCount', { count: pillars.length })}
+                heading={_ts('analysis', 'pillarAnalysisCount', { count: pillars?.length })}
```
Comment (AdityaKhatri) [inline app/views/AnalysisDashboard/Analysis/index.tsx]: Lets remove _ts and replace this directly with a text.

File: app/views/AnalysisDashboard/index.tsx [hunk 7]
```diff
@@ -73,33 +179,27 @@ const colorScheme = [
...
+const labelSelector = (item: AuthoringOrganizations) => item.title;
 const valueSelector = (item: AuthoringOrganizations) => item.count;
 const tickFormatter = (title: string) => title;
 
-const timelineLabelSelector = (item: TimelineData) => item.label;
-const timelineValueSelector = (item: TimelineData) => item.value;
-const timelineKeySelector = (item: TimelineData) => item.key;
+const timelineLabelSelector = (d: TimelineData): string => d.label;
+const timelineValueSelector = (d: TimelineData): number => d.value;
+const timelineKeySelector = (d: TimelineData): string => d.key;
```
Comment (AdityaKhatri) [inline app/views/AnalysisDashboard/index.tsx]: why replace the d?

File: app/views/AnalysisDashboard/index.tsx [hunk 8]
```diff
@@ -196,64 +325,92 @@ function AnalysisDashboard(props: AnalysisDashboardProps) {
...
+    const analyzedEntriesCount = analysisSummaryData
+        ?.project?.analysisOverview?.analyzedEntriesCount;
+    const analyzedLeadsCount = analysisSummaryData
+        ?.project?.analysisOverview?.analyzedLeadsCount;
+    const totalEntriesCount = analysisSummaryData
+        ?.project?.analysisOverview?.totalEntriesCount;
+    const totalLeadsCount = analysisSummaryData
+        ?.project?.analysisOverview?.totalLeadsCount;
+
+    const analysisRendererParams = useCallback((id: string, data: AnalysisSummaryType) => ({
```
Comment (AdityaKhatri) [inline app/views/AnalysisDashboard/index.tsx]: Add return time in renderer params for better typecheck

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/AnalysisDashboard/index.tsx [hunk 9]
```diff
@@ -139,122 +234,163 @@ function AnalysisDashboard(props: AnalysisDashboardProps) {
...
-
     const handleAnalysisEditSuccess = useCallback(() => {
-        triggerGetAnalysis();
-        retriggerAnalysisOverview();
+        getProjectAnalysis();
         setModalHidden();
-    }, [setModalHidden, triggerGetAnalysis, retriggerAnalysisOverview]);
+    }, [setModalHidden, getProjectAnalysis]);
 
-    const handleAnalysisEditClick = useCallback((analysisId: number | undefined) => {
```
Comment (AdityaKhatri) [inline app/views/AnalysisDashboard/index.tsx:221]: Please do not undo this

File: app/views/AnalysisDashboard/index.tsx [hunk 10]
```diff
@@ -73,56 +179,45 @@ const colorScheme = [
...
-    entriesTotal: number;
-    analyzedEntriesCount: number;
-    sourcesTotal: number;
-    analyzedSourceCount: number;
-    authoringOrganizations: AuthoringOrganizations[];
-}
+
 interface AnalysisDashboardProps {
     className?: string;
+    analysisId: string;
```
Comment (AdityaKhatri) [inline app/views/AnalysisDashboard/index.tsx]: why? Please no

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Review by shreeyash07 — APPROVED** [review]

**Pull request merged**